packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -302,16 +311,33 @@ export class EngineReportingAgent<TContext = any> {
     if (protobufError) {
       throw new Error(`Error encoding trace: ${protobufError}`);
     }
-    const encodedTrace = Trace.encode(trace).finish();
 
-    const signature = await this.getTraceSignature({
-      queryHash,
-      documentAST,
-      queryString,
-      operationName,
-    });
+    let statsReportKey: string;
+    // It's important to check parse failure first, since parse failures always
+    // indicate validation failure
+    if (!documentAST) {
+      statsReportKey = `## GraphQLParseFailure`;
+      if (this.options.sendOperationDocumentsOnUnexecutableOperation && queryString) {
+        trace.unexecutedOperationBody = queryString;
+        trace.unexecutedOperationName = operationName;
+      }
+    } else if (gqlValidationFailure) {
+      statsReportKey = `## GraphQLFailure`;
+      if (this.options.sendOperationDocumentsOnUnexecutableOperation && queryString) {
+        trace.unexecutedOperationName = queryString;
+        trace.unexecutedOperationName = operationName;
+      }
+    } else {
```

Comment:
Do we want to handle the case of "explicit operation name does not match an operation" somehow, which we discovered in registry-server-land is not quite a validation failure but also stops execution?   I guess the `execute` phase technically starts but it doesn't actually execute anything.

Reply:
Meh. If you have any ideas for how we can support that without mucking around with the request pipeline, I'd be happy to fit it into this PR, but otherwise I'd like to address it as a follow-up.